### DISCUSSION TITLE
Array and Linked list簡介

### DISCUSSION THREAD

**Discussion by Tomlord1122** [1101-GS4719-C/Python-Community-GS4719-C#196] [Ideas]
這邊介紹一下兩種儲存資料的資料結構
Array我想大家比較熟悉，是宣告一個大小，然後可以在裡面insert或delete資料。
其優點是他可以支援隨機存取，但缺點顯而易見的是他在insert delete的時間複雜度比linked list高，因為他需要移動其他格子的資料。

linked list 則是宣告一個node，我們稱他為節點，而一個節點裡面會有兩個資訊，一個是node 的 data，另一個則是pointer，指向下一個節點，如果沒有下一個節點，則是null，linked list 的優點不用事先宣告大小，且insert delete 的時間複雜度比array還要小，但缺點是無法進行隨機存取，只能用sequential access的方式，資料上的保存也比較有風險，因為在一條搜尋路徑上，如果有一個pointer斷掉，就找不到之後的data了。

**Comment by HSU-YU-JEN** [comment]
很實用的分享~想補充一下linked list 的使用情況大概於
1.無法預期資料數量時，使用Linked list就沒有resize的問題。
2.需要頻繁地新增/刪除資料時。
3.不需要快速查詢資料。
![image](https://user-images.githubusercontent.com/90545550/146289130-47219d3f-5087-43a5-b6b3-aa0de2dde590.png)

**Comment by swerfg03joe** [comment]
array 和 link list 是資料結構的基礎，兩者各有其優缺點，設計者要能正確分析一件工作到底要選擇什麼樣的儲存方式，因資料儲存方法不同，對整個程式碼的布局也有相當大的影響，倒不是說寫不出來，而是選擇適當的結構可以簡化問題，另外還能追求執行速度
